### PR TITLE
Replace `ensure-protocol-specified-in-site-url` with a Liquibase migration

### PR DESCRIPTION
**Pull request by camsaul**
Part of #20449

### CODE REVIEW

File: src/metabase/db/data_migrations.clj [hunk 1]
```diff
@@ -178,7 +167,7 @@
 (defmigration ^{:author "senior", :added "0.30.0"} clear-ldap-user-local-passwords
   (db/transaction
     (doseq [user (db/select [User :id :password_salt] :ldap_auth [:= true])]
-      (db/update! User (u/the-id user) :password (creds/hash-bcrypt (str (:password_salt user) (UUID/randomUUID)))))))
+      (db/update! User (u/the-id user) :password (creds/hash-bcrypt (str (:password_salt user) (java.util.UUID/randomUUID)))))))
```
Comment (noahmoss) [inline src/metabase/db/data_migrations.clj:170]: Is this change intentional? (And if so, why?) Seems like it's done in every one of your migration PRs.

**Review by noahmoss — APPROVED** [review]

File: src/metabase/db/data_migrations.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/db/data_migrations.clj:170]: TBH I only did it because the new version of `clj-refactor` keeps trying to change `(:import java.util.UUID)` to `(:import (java.util UUID))` when you run `cljr-clean-namespace` and I got sick of changing it back by hand in 8 different branches. It gets removed anyway in the end
